Solutions/Azure Activity/Analytic Rules/Creating_Anomalous_Number_Of_Resources_detection.yaml

```diff
@@ -22,43 +20,30 @@ query: |
-  by ResourceId, Caller, OperationNameValue, Resource, ResourceGroup
-  | mvexpand CallerIpAddress
-  | where isnotempty(CallerIpAddress)
-  | make-series dResourceCount=dcount(ResourceId)  default=0 on StartTimeUtc in range(startofday(ago(7d)), now(), 1d)
-  by Caller, tostring(ActivityTimeStamp), tostring(ActivityStatusValue), tostring(OperationIds), tostring(CallerIpAddress), tostring(CorrelationId), ResourceId, OperationNameValue , Resource, ResourceGroup
-  | extend (RSquare,Slope,Variance,RVariance,Interception,LineFit)=series_fit_line(dResourceCount)
-  | where Slope > 0.2
-  | join kind=leftsemi (
-  // Last day's activity is anomalous
-  AzureActivity
+  | where TimeGenerated between (startofday(ago(starttime)) .. startofday(now()))
+  | where OperationNameValue in~ (szOperationNames)
+  | project TimeGenerated, Caller 
+  | make-series Total = count() on TimeGenerated from startofday(ago(starttime)) to startofday(now()) step timeframe by Caller; 
+  TimeSeriesData
+  | extend (anomalies, score, baseline) = series_decompose_anomalies(Total, 3, -1, 'linefit')
+  | mv-expand Total to typeof(double), TimeGenerated to typeof(datetime), anomalies to typeof(double), score to typeof(double), baseline to typeof(long) 
   | where TimeGenerated >= startofday(ago(endtime))
+  | where anomalies > 0 and baseline > 0
+  | project Caller, TimeGenerated, Total, baseline, anomalies, score
+  | join (AzureActivity
+  | where TimeGenerated > startofday(ago(endtime)) 
   | where OperationNameValue in~ (szOperationNames)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ActivityTimeStamp = makelist(TimeGenerated), ActivityStatusValue = makelist(ActivityStatusValue),
-  OperationIds = makelist(OperationId), CallerIpAddress = makelist(CallerIpAddress), CorrelationId = makelist(CorrelationId)
-  by ResourceId, Caller, OperationNameValue, Resource, ResourceGroup
-  | mvexpand CallerIpAddress
-  | where isnotempty(CallerIpAddress)
-  | make-series dResourceCount=dcount(ResourceId)  default=0 on StartTimeUtc in range(startofday(ago(1d)), now(), 1d)
-  by Caller, tostring(ActivityTimeStamp), tostring(ActivityStatusValue), tostring(OperationIds), tostring(CallerIpAddress), tostring(CorrelationId), ResourceId, OperationNameValue , Resource, ResourceGroup
-  | extend (RSquare,Slope,Variance,RVariance,Interception,LineFit)=series_fit_line(dResourceCount)
-  | where Slope > 0.2
-  ) on Caller, CallerIpAddress
-  | mvexpand todynamic(ActivityTimeStamp), todynamic(ActivityStatusValue), todynamic(OperationIds), todynamic(CorrelationId)
-  | extend timestamp = ActivityTimeStamp, AccountCustomEntity = Caller, IPCustomEntity = CallerIpAddress
+  | summarize count(), make_set(OperationNameValue), make_set(_ResourceId) by bin(TimeGenerated, timeframe), Caller) on TimeGenerated, Caller 
+  | extend Name = tostring(split(Caller,'@',0)[0]), UPNSuffix = tostring(split(Caller,'@',1)[0])
```

Comment:
This might not always be a UPN, it could also be an SP ID.

Reply:
correct. In those cases SP ID would be mapped to Name and UPNSuffix would be null

Reply:
What I would do in this case is extend SP name with an iif statement so it's a separate field, that way you can project it as an entity of a different type (AAD User Id)
for examples
| extend SPID = iif(Caller !has "@", Caller, "")

Reply:
Thankyou Pete for the suggestion. I've applied the same.